internal/auth/clusterapitoken.go

```diff
@@ -87,19 +144,91 @@ func JwtTokenWithM2M(ctx context.Context, ttl *time.Duration) (string, error) {
 
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 	client := &http.Client{Timeout: 10 * time.Second}
+
+	// attempt credential refresh & retry once
+	accessToken, retryable, err := doM2MTokenRequest(client, req)
+	if err != nil && retryable {
+		if errRef := ensureM2MCredentials(ctx, true); errRef == nil {
+			credsMu.Lock()
+			clientID, clientSecret = cachedClientID, cachedClientSecret
+			credsMu.Unlock()
+
+			data.Set("client_id", clientID)
+			data.Set("client_secret", clientSecret)
+
+			req2, r2err := http.NewRequestWithContext(ctx, "POST", tokenURL, bytes.NewBufferString(data.Encode()))
+			if r2err != nil {
+				return "", fmt.Errorf("failed to create retry token request: %w", r2err)
+			}
+
+			req2.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+			client = &http.Client{Timeout: 10 * time.Second}
+
+			return doFinalTokenRequest(client, req2)
+		}
+	}
```

Comment:
This might be cleaner with a backoff retry, would remove the need for two separate request functions and a retryable boolean.